src/opml_urlreader.cpp

```diff
@@ -0,0 +1,116 @@
+#include "opml_urlreader.h"
+
+#include <cstring>
+
+#include "utils.h"
+
+namespace newsboat {
+
+opml_urlreader::opml_urlreader(configcontainer* c)
+	: cfg(c)
+{
+}
+
+void opml_urlreader::write_config()
+{
+	// do nothing.
+}
+
+void opml_urlreader::reload()
+{
+	urls.clear();
+	tags.clear();
+	alltags.clear();
+
+	std::vector<std::string> urls =
+		utils::tokenize_quoted(this->get_source(), " ");
+
+	for (const auto& url : urls) {
+		LOG(level::DEBUG,
+			"opml_urlreader::reload: downloading `%s'",
+			url);
+		std::string urlcontent = utils::retrieve_url(url, cfg);
+
+		xmlDoc* doc =
+			xmlParseMemory(urlcontent.c_str(), urlcontent.length());
+
+		if (doc == nullptr) {
+			LOG(level::ERROR,
+				"opml_urlreader::reload: parsing XML file "
```

Comment:
Does it make sense to include the URL a well? The log above containing the URL is just of debug level.